src/screens/Search/AlgoSearchScreen.js

```diff
@@ -0,0 +1,220 @@
+import { TabViewAnimated, TabBar } from 'react-native-tab-view';
+import { AlgoResultsList, TopsList } from './Lists';
+import * as colors from 'kitsu/constants/colors';
+
+const ALGOLIA_ID = 'AWQO5J657S';
+
+const styles = {
+  container: {
+    backgroundColor: colors.listBackPurple,
+    flex: 1,
+  },
+  scrollView: {
+    backgroundColor: colors.listBackPurple,
+  },
+  searchBoxContainer: {
+    height: 36,
+    backgroundColor: colors.white,
+    paddingLeft: 14,
+    paddingRight: 14,
+    borderColor: colors.imageGrey,
+    borderBottomWidth: StyleSheet.hairlineWidth,
+    marginLeft: 9,
+    marginRight: 9,
+    borderRadius: 2,
+  },
+  searchBoxInput: {
+    fontSize: 13,
+    fontFamily: 'OpenSans',
+    color: colors.placeholderGrey,
+    alignSelf: 'center',
+    textAlign: 'center',
+  },
+  searchBoxIcon: {
+    color: '#9D9D9D',
+    fontSize: 17,
+    alignItems: 'center',
+    marginTop: 5,
+  },
+  tabBar: {
+    backgroundColor: colors.listBackPurple,
+    borderTopWidth: 0,
+    borderBottomWidth: 0,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    height: 40,
+    paddingRight: 5,
+    paddingLeft: 5,
+    shadowColor: 'black',
+    shadowOpacity: 0.1,
+    shadowRadius: StyleSheet.hairlineWidth,
+    marginRight: 28,
+    marginLeft: 28,
+  },
+  tabBarItem: {
+    height: 27,
+    marginTop: 0,
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderBottomWidth: 0,
+    borderTopWidth: 0,
+    borderLeftWidth: 0,
+    borderRightWidth: 0,
+  },
+  tabBarTextActive: {
+    color: colors.tabbarSelectedTextColor,
+    fontFamily: 'OpenSans',
+    fontWeight: '600',
+    opacity: 1,
+    fontSize: 12,
+  },
+  tabBarText: {
+    color: '#ffffff',
+    fontFamily: 'OpenSans',
+    fontWeight: '600',
+    opacity: 0.6,
+    fontSize: 12,
+  },
+};
+
+const SearchBox = connectSearchBox(({ refine, currentRefinement, placeholder }) => (
+  <Item style={styles.searchBoxContainer}>
+    <Icon name="ios-search" style={styles.searchBoxIcon} />
+    <Input
+      placeholder={placeholder}
+      value={currentRefinement}
+      onChangeText={t => refine(t)}
+      style={styles.searchBoxInput}
+      placeholderTextColor={colors.placeholderGrey}
+    />
+  </Item>
+));
+
+const Hits = connectInfiniteHits(AlgoResultsList);
+
+class AlgoSearchScreen extends Component {
+  static navigationOptions = {
+    title: 'Search',
+    headerStyle: {
+      backgroundColor: colors.darkPurple,
+      shadowOpacity: 0,
+      height: 0,
+    },
+    tabBarIcon: ({ tintColor }) => (
+      <Icon ios="ios-search" android="md-search" style={{ fontSize: 24, color: tintColor }} />
+    ),
+  };
+
+  state = {
+    query: undefined,
+    index: 0,
+    routes: [
+      {
+        key: '1',
+        title: 'Anime',
+        apiKey: 'MTc1MWYzYzNiMjVjZjM5OWFiMDc1YWE4NTNkYWMyZjE4NTk2YjkyNjM1YWJkOWIwZTEwM2U1YmUyMjgyODIwY3Jlc3RyaWN0SW5kaWNlcz1wcm9kdWN0aW9uX21lZGlhJmZpbHRlcnM9',
```

Comment:
I don't think so, I incline towards using for config management https://github.com/luggit/react-native-config. However, the absence of any such library in the architecture is why I committed it. I believe in a different task we should migrate all the api keys to `react-native-config`